### ISSUE TITLE
Not really a multi-tenant SaaS

### ISSUE THREAD

**Issue by PavelPikat** [Azure-Samples/active-directory-dotnet-webapi-multitenant-windows-store#7]
I want to expose the sample API to 3rd party clients and their apps registered in their Azure AD tenants (so basically API is SaaS)

First, it is only possible to assign delegated permissions to API after the admin of a 3rd party client application consented to API.

Second, the API is unable to dynamically change its manifest and add all external 3rd party client application as "knownClientApplications".

Even if I manually consent to the Web API from a 3rd party tenant, I get access denied by calling it (even though I acquire the access token).

Seems like it's impossible to protect the Web API with Azure AD without attaching a web\native application from the same tenant to it.

So how do we build truly SaaS experience and expose Web API to external consumers and partners?

**Comment by WayPoint7** [comment]
For what it's worth, I give my experience with this example and what appear to me as comparable problems we encountered by exposing our API to 3rd parties.
We have eventually found out that it was necessary to add a setting in AD, that is nowhere mentioned or talked about: give your API access to read the AD directory data:
go to the classic portal, choose your tenant, choose your api application, select Configure and add the setting "permission to other applications" to also "Read directory data". For us, this worked!
Good luck!
